### ISSUE TITLE
support for "other" vpn systems (enhancement request)

### ISSUE THREAD

**Issue by nicklan** [dlenski/vpn-slice#71]
It would be really cool if this script could reconfigure routing after it's been set up by a different vpn solution. I imagine this would involve inspecting the current routing table, and then deleting/creating routes as needed.

I'd be interested in looking into how to do this, but don't currently have enough knowledge about how all the routing etc works to know what the right way to do it would be. If you have some pointers for where I could get up to speed with that, that would be great.

Thanks!

**Comment by dlenski** [comment]
> It would be really cool if this script could reconfigure routing after it's been set up by a different vpn solution. I imagine this would involve inspecting the current routing table, and then deleting/creating routes as needed.

I've thought about this too but… _which_ different VPN solutions?

Other open-source VPNs like OpenVPN should generally be amenable to routing customization. (I even hacked together a horrible kludged version of vpn-slice to work with screen-scraped output from [sstp-client](http://sstp-client.sourceforge.net)+pppd a few years back.)

Closed-source VPNs will generally make this difficult-or-impossible. Many of them try to monitor for changes to the routing table and interfere with them. “Securiteh.”

> I'd be interested in looking into how to do this, but don't currently have enough knowledge about how all the routing etc works to know what the right way to do it would be. If you have some pointers for where I could get up to speed with that, that would be great.

Basically everything I know about routing comes from reading the stuff described in the [inspiration and credits section](https://github.com/dlenski/vpn-slice#inspiration-and-credits), writing `vpn-slice` (originally motivated by a need at my previous job), and contributing to OpenConnect.

**Comment by nicklan** [comment]
> Closed-source VPNs will generally make this difficult-or-impossible. Many of them try to monitor for changes to the routing table and interfere with them. “Securiteh.”

Yeah, that's the trickiest bit I guess. I suppose you could workaround most anything, but it would be very specific to each tool.

Anyway, thanks for the pointers, I'll go do some reading. Feel free to close this, or leave it open if you want a place to continue discussion around the topic.

**Comment by mschilli87** [comment]
Personally I'd love to use vpn-slice with OpenVPN.

**Comment by dlenski** [comment]
> 
Personally I'd love to use vpn-slice with $X.

I've said it before and I'll say it again :smile: 

"I don't use $X myself, but well-designed pull requests to support $X would be welcome, and I can certainly help review and clean them up."
